### PR TITLE
add recipe for basedpyright

### PR DESCRIPTION
**Pull request by lucascolley**
https://github.com/DetachHead/basedpyright/issues/410

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by lucascolley** [comment]
@conda-forge/help-nodejs grayskull generated a recipe with `nodejs-wheel` - what do we need to do to get that working?

**Comment by danielnachun** [comment]
I made a comment in the linked issue here: https://github.com/DetachHead/basedpyright/issues/410#issuecomment-2502751988.  I'm hoping the upstream developer is open to providing an option for users to just an existing NodeJS installation instead of vendoring it with `nodejs-wheel`.

**Comment by xhochy** [comment]
For starters, you could patch out the dependency from `pyproject.toml` and add a run dependency on `nodejs >=20`. That would be OK to get it working.

**Comment by lucascolley** [comment]
`docify >=1.0.0  does not exist`

**Comment by lucascolley** [comment]
thanks @xhochy, looks like that worked!

@detachhead would you like to be added as a maintainer here?

**Comment by DetachHead** [comment]
sure, though i wont be much help since i don't use conda

**Comment by lucascolley** [comment]
@conda-forge/help-nodejs @conda-forge/help-python ready for review, thanks! :)

**Comment by lucascolley** [comment]
Hmm, I'm sure the patch succeeded in CI once, but is now failing to be applied.

**Comment by ocefpaf** [comment]
> For starters, you could patch out the dependency from `pyproject.toml` and add a run dependency on `nodejs >=20`. That would be OK to get it working.

@xhochy nodejs_wheel doesn't seem to be a wheel for nodejs but rather a pure python wrapper to call an installed node binary.

@lucascolley I guess you'll have to `nodejs_wheel` as a package and drop the patch.

**Comment by njzjz** [comment]
Hi this is the author of `nodejs-wheel`. I could contribute a recipe to conda-forge if it is needed by `basedpyright`.

> nodejs_wheel doesn't seem to be a wheel for nodejs but rather a pure python wrapper to call an installed node binary.

The wheel part and the python wrapper (`nodejs_wheel` module) are both in the `nodejs-wheel-binaries` package. `nodejs-wheel` is an empty package which depends on `nodejs-wheel-binaries` but adds entry points (`node`, `npm`, `npx`) which might be conflict with system's already installed nodejs. The GH repo contains `pyproject.toml` for both packages (one in the root directory and the other in the `cmd` directory).

**Comment by lucascolley** [comment]
> I could contribute a recipe to conda-forge if it is needed by basedpyright

sounds like that would be useful, thanks!

**Comment by njzjz** [comment]
> I could contribute a recipe to conda-forge if it is needed by `basedpyright`.

#28400 is ready for review.

File: recipes/basedpyright/meta.yaml [hunk 1]
```diff
@@ -24,12 +22,14 @@ requirements:
   host:
     - python {{ python_min }}
     - pdm-backend >=2.3.0
+    - nodejs-wheel >=20.13.1
     - typing_extensions >=4.12.2
     - docify >=1.0.0
     - pip
     - nodejs >=20
```
Comment (ocefpaf) [inline recipes/basedpyright/meta.yaml:29]: The nodejs-wheel package should bring this to you. 

```suggestion
```

File: recipes/basedpyright/meta.yaml [hunk 2]
```diff
@@ -24,12 +22,14 @@ requirements:
...
   run:
     - python >={{ python_min }}
+    - nodejs-wheel-binaries >=20.13.1
```
Comment (ocefpaf) [inline recipes/basedpyright/meta.yaml:32]: I believe it is needed only to build, right?

File: recipes/basedpyright/meta.yaml (hunk 1 above)
Comment (lucascolley) [reply recipes/basedpyright/meta.yaml:29]: Oops, `git pull --rebase` failure on my part

File: recipes/basedpyright/meta.yaml (hunk 2 above)
Comment (ocefpaf) [reply recipes/basedpyright/meta.yaml:31]: I may be wrong... Let's see what happens.

**Comment by njzjz** [comment]
> @njzjz would you be able to contribute a package for `nodejs-wheel-binaries` as well? It seems like that is a runtime dependency here

Please use `nodejs-wheel` directly

**Comment by lucascolley** [comment]
@conda-forge/help-nodejs, @conda-forge/help-python CI passing, ready for review, thanks!!

**Review by xhochy — APPROVED** [review]

**Comment by lucascolley** [comment]
thank you all!!

**Pull request merged**